bool see_ball
bool see_bucket
int32[2] ball_pos
int32[2] bucket_pos
int32 ball_size
int32 bucket_size
